Client/src/components/game/game.tsx

```diff
@@ -0,0 +1,298 @@
+import '../../css/game.css';
+import Maze from './maze';
+import RightBar from './rightBar';
+import LeftBar from './leftBar';
+import { useEffect, useRef, useState } from 'react';
+import { animal, MazeTileType, minionType, TowerType } from '../../utils/types';
+import { Graph, value } from '../../utils/graph';
+import { aStar, distanceConstruct, getDirection, vBFS, vDFS, vDijk } from '../../utils/path-finding-algo';
+import { bubbleSortAlgo, insertionSortAlgo, mergeSortAlgo, quickSortAlgo, selectionSortAlgo } from '../../utils/sorting-algo';
+import { uniqueNamesGenerator, Config, names} from 'unique-names-generator'
```

Comment:
I see two possible hooks here, when requesting visuals and when requesting animations